File: Task-011/wordCount_Spark.py
import sys
 
from pyspark import SparkContext
from pyspark.streaming import StreamingContext
from pyspark.sql import SparkSession
 
if __name__ == "__main__":
	
	# create Spark context with necessary configuration
	sc = SparkContext("local[*]","PySpark Word Count Exmaple")
	ssc = StreamingContext(sc, 3)

	ss = SparkSession.builder \
	.enableHiveSupport() \
	.getOrCreate()

	
	# read data from text file and split each line into words; read file into RDD
	words = sc.textFile("/home/fieldemployee/Shakespeare.txt").flatMap(lambda line: line.split(" "))
	
	# count the occurrence of each word
	wordCounts = words.map(lambda word: (word, 1)).reduceByKey(lambda a,b:a +b)
	
	# save the counts to output
	wordCounts.saveAsTextFile("/home/fieldemployee/output/")

	ssc.start()
	ssc.awaitTermination()

File: Task-012/wordCount_Spark.scala

import org.apache.spark._
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._
import org.apache.spark.SparkConf

object WordCount{
        def main(args:Array[String]){
                val conf = new SparkConf()
                conf.set("spark.master", "local")
                conf.set("spark.app.name", "App")
                val sc = new SparkContext(conf)
                var map = sc.textFile("/home/fieldemployee/Shakespeare.txt").flatMap(line => line.split(" ")).map(word => (word,1));
                var counts = map.reduceByKey(_ + _);
                counts.saveAsTextFile("/home/fieldemployee/wordCount_Spark_output/");
            
                sc.stop()
    